functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
 #GET API method function
 def get(param1, param2):
     #connects to db
+    logging.debug('Attempting a db conenction')
     cnxn = connect()
-    cursor = cnxn.cursor()
-    logging.info('opened connection')        
-    logging.info('Going to execute select query')
+    cursor = cnxn.cursor() 
     try:
+        logging.debug('opened connection')        
+        logging.debug('Attempting to execute GET task query')
         #Get task title, description and user name by userId and taskId
-        sql_query = ("""SELECT tasks.title, tasks.description, CONCAT (users.firstName, ' ', users.lastName) AS "user" 
-        FROM [dbo].[tasks] JOIN [dbo].[users] 
-        on [dbo].[tasks].userId = [dbo].[users].userId
-        WHERE [dbo].[users].userId = ? AND [dbo].[tasks].taskId = ?""")
+        sql_query = ("""SELECT tasks.title, tasks.description, tasks.createdDate, tasks.dueDate, 
+                    tasks.completed, tasks.completedDate, CONCAT (users.firstName, ' ', users.lastName) AS "user" 
+                    FROM [dbo].[tasks] JOIN [dbo].[users] 
+                    on [dbo].[tasks].userId = [dbo].[users].userId
+                    WHERE [dbo].[users].userId = ? AND [dbo].[tasks].taskId = ?""")
         cursor.execute(sql_query, param1, param2)
-        logging.info('Executed the query')         
-        data = cursor.fetchall()
-        logging.info(f"Got result: {data}")
-        return data
-#irrespective of results, closes connection
+        logging.debug('Executed the query')          
+        row = cursor.fetchone()
+        logging.info(f"Got result: {row}")
+        if not row:
+            logging.error('No record with the requested parameters')
+            return func.HttpResponse('Task not found', status_code=500)
+        else:
+            columns = [column[0] for column in cursor.description]
+            data = dict(zip(columns, row))
+        return func.HttpResponse(json.dumps(data, default=str), status_code=200, mimetype="application/json")
     finally:
         cursor.close()
         cnxn.close()
-        logging.info('Closed the connection')
+        logging.info('Closed the db connection')   
 
-#POST API method function       
-def post(param):

```

Comment:
don't we need to implement POST for this function?

Reply:
Not for this function. The taskId needs to be in place for users/user_id/tasks/task_id in order for the endpoint to be called. it is disabled in the swagger doc from the last quarter, too. ;) Creation should happen in the users/user_id/tasks/ endpoint

Reply:
thanks a lot for the explanation. I was thinking that might be a good idea to return a 405 error when we get a request that is not GET, DELETE, or PUT.